src/level/level.cpp

```diff
@@ -238,22 +247,112 @@ namespace level {
         artifacts_.remove(i);
     }
 
+    void Level::enemyShoot(entities::Shooter *s, manager::Level *levelManager){
+        Collidable *collision;
+        bool firable = true;
+        if(s->canShoot()){
+            if(s->getPosition().riga == levelManager->getPlayer()->getPosition().riga){
+                if(s->getPosition().colonna < levelManager->getPlayer()->getPosition().colonna){
+                    for(int i = (s->getPosition().colonna + 1); i <  (levelManager->getPlayer()->getPosition().colonna); i++){
+                        collision = getCollision(Position (s->getPosition().riga, i), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
+                    if(firable){
+                        s->resetShootCoolDown();
+                        logger_.info("enemy firing a bullet");
+                        Position bulletPosition = Position(s->getPosition().riga, s->getPosition().colonna + 1);
+                        weapon::Bullet *bullet = new weapon::Bullet(bulletPosition, enums::Direction::RIGHT, s->getAttack());
+                        this->addBullet(bullet);
+                    }
+                }
+                else{
+                    for(int i = (s->getPosition().colonna - 1); i >  (levelManager->getPlayer()->getPosition().colonna); i--){
+                        collision = getCollision(Position (s->getPosition().riga, i), levelManager);
+                        enums::CollisionType type = enums::CollisionType::NONE;
+                        if (collision != nullptr) type = collision->getCollisionType();
+                        if(type != enums::CollisionType::NONE){
+                            firable = false;
+                        }
+                    }
+                    if(firable){
+                        s->resetShootCoolDown();
+                        logger_.info("enemy firing a bullet");
+                        Position bulletPosition = Position(s->getPosition().riga, s->getPosition().colonna - 1);
+                        weapon::Bullet *bullet = new weapon::Bullet(bulletPosition, enums::Direction::LEFT, s->getAttack());
+                        this->addBullet(bullet);
+                    }
+                }
+            }
+            if(s->getPosition().colonna == levelManager->getPlayer()->getPosition().colonna){
```

Comment:
questi forse possono essere gestiti in maniera più generale, i due if fanno la stessa cosa solo in direzione differente